GliaWidgets/Sources/AlertManager/AlertInputType.swift

```diff
@@ -0,0 +1,68 @@
+import Foundation
+
+enum AlertInputType: Equatable {
+    case error(error: (any Error)?, dismissed: () -> Void)
+    case cameraSettings(dismissed: () -> Void)
+    case endEngagement(confirmed: () -> Void)
+    case endScreenShare(confirmed: () -> Void)
+    case leaveQueue(confirmed: () -> Void)
+    case liveObservationConfirmation(
+        link: (WebViewController.Link) -> Void,
+        accepted: () -> Void,
+        declined: () -> Void
+    )
+    case mediaSourceNotAvailable(dismissed: () -> Void)
+    case microphoneSettings(dismissed: () -> Void)
+    case operatorEndedEngagement(action: () -> Void)
+    case unsupportedGvaBroadcastError(dismissed: () -> Void)
+    case unavailableMessageCenter(dismissed: () -> Void)
+    case unavailableMessageCenterForBeingUnauthenticated(dismissed: () -> Void)
+    case mediaUpgrade(
+        operators: String,
+        offer: CoreSdkClient.MediaUpgradeOffer,
+        accepted: (() -> Void)? = nil,
+        declined: (() -> Void)? = nil,
+        answer: CoreSdkClient.AnswerWithSuccessBlock
+    )
+    case screenSharing(
+        operators: String,
+        accepted: (() -> Void)? = nil,
+        declined: (() -> Void)? = nil,
+        answer: CoreSdkClient.AnswerBlock
+    )
+
+    static func == (lhs: AlertInputType, rhs: AlertInputType) -> Bool {
+        switch (lhs, rhs) {
+        case (.cameraSettings, .cameraSettings):
+            return true
+        case (.endEngagement, .endEngagement):
+            return true
+        case (.endScreenShare, .endScreenShare):
+            return true
+        case (.leaveQueue, .leaveQueue):
+            return true
+        case (.liveObservationConfirmation, .liveObservationConfirmation):
+            return true
+        case (.mediaSourceNotAvailable, .mediaSourceNotAvailable):
+            return true
+        case (.microphoneSettings, .microphoneSettings):
+            return true
+        case (.operatorEndedEngagement, .operatorEndedEngagement):
+            return true
+        case (.unsupportedGvaBroadcastError, .unsupportedGvaBroadcastError):
+            return true
+        case (.unavailableMessageCenter, .unavailableMessageCenter):
+            return true
+        case (.unavailableMessageCenterForBeingUnauthenticated, .unavailableMessageCenterForBeingUnauthenticated):
+            return true
+        case (.mediaUpgrade, .mediaUpgrade):
+            return false
```

Comment:
I think it would be nice to compare `operators` & `offer`, since a few media upgrades may happen at the same time. should we?

Reply:
I returned them as false(not the same) regardless if they are actually different or not. The reason:

Let's say media upgrade is sent and then right after that another same media upgrade is sent. Since the same alerts are filtered/blocked, the most recent one is not shown. If I accept the offer, then nothing happens because GliaHub has already cancelled that request, and waits for the response on the most recent one. That is why we can't filter them out. And the same goes for screensharing offer.

Reply:
agree with everything your said and I suppose if two the same alerts come to this == function they will not be filtered out because they all fully the same, so the same alert may appear twice. So, would it make sense to compare `operators` and `offer` also to remove duplications in filter function?

Reply:
I don't think it makes sense to compare `operators` and `offer` because we intentionally don't want to filter out duplications. The reason: If we do filter out the most recent one (i.e mediaUpgrade), then we are displaying the old one. If we accept the old one, then nothing happens on GliaHub side, because Gliahub only cares about the latest request. To guarantee that the latest is always shown, we have to let through all mediaUpgrades and screenShareOffers in order to guarantee proper functionality.

Reply:
If AlertManager is called to present `mediaUpgrade` from several places with the same `offer` that will help to not show it multiple times. Or I misunderstood something?

Reply:
`mediaUpgrade` can only be called to present via interactor which is technically one place. The idea of filtering some alerts out is for all the other alert types, but more specifically for errors, such as 409 error.